src/drivers/drv_spi_bmi270.c

```diff
@@ -145,4 +152,492 @@ void bmi270_read_data(uint8_t reg, uint8_t *data, uint32_t size) {
       spi_make_seg_buffer(data, NULL, size),
   };
   spi_seg_submit_wait(&gyro_bus, segs);
-}
\ No newline at end of file
+}
+
+static int8_t bmi270_compute_gyro_cas(uint8_t raw) {
+  const uint8_t masked = raw & BMI270_GYRO_CAS_MASK;
+  if ((masked & BMI270_GYRO_CAS_SIGN_BIT_MASK) == 0) {
+    return (int8_t)(masked);
+  }
+  return (int8_t)(masked - 128);
+}
+
+void bmi270_read_gyro_data(gyro_data_t *data) {
+  uint8_t buf[14];
+
+  const spi_txn_segment_t gyro_segs[] = {
+      spi_make_seg_const(BMI270_REG_ACC_DATA_X_LSB | 0x80),
+      spi_make_seg_const(0xFF),
+      spi_make_seg_buffer(buf, NULL, 12),
+  };
+  spi_seg_submit(&gyro_bus, NULL, gyro_segs);
+
+  const spi_txn_segment_t cas_segs[] = {
+      spi_make_seg_const(BMI270_REG_FEATURES_0_GYR_CAS | 0x80),
+      spi_make_seg_const(0xFF),
+      spi_make_seg_buffer(buf + 12, NULL, 2),
+  };
+  spi_seg_submit(&gyro_bus, NULL, cas_segs);
+
+  spi_txn_wait(&gyro_bus);
```

Comment:
I was wondering would this cas_segs contains two pull down on CS pin

Reply:
each `spi_seg_submit` here represents one spi transaction with one pull down.

Reply:
then no problem. you are reading the correct reg